common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -101,9 +114,31 @@ class HTTPClient(
             connection.disconnect()
         }
 
-        val responseBody = payload?.let { JSONObject(it) } ?: throw IOException("Network call payload is null.")
         log(LogIntent.DEBUG, NetworkStrings.API_REQUEST_COMPLETED.format(connection.requestMethod, path, responseCode))
-        return Result(responseCode, responseBody)
+        if (payload == null) {
+            throw IOException("Network call payload is null.")
+        }
+        val result = HTTPResult(responseCode, payload)
+        val processedResponse = eTagManager.processResponse(httpRequest, connection, result)
```

Comment:
it took me a bit to figure out exactly what `processResponse` means in this context, but really it's "get the response from storage if needed, else get the response from the request", right? Its current name sort of implies that we're doing something _to_ the response, i.e.: processing it in some way. 
Could we come up with a name that's more reflective of what it does? 

i.e.: `retrieveResponsePayload` or something to that effect?

Reply:
I agree, I am not a fan of `processResponse` either. The thing is that the function will "retrieve" from cache only if the reponse is 304, but it will return what's returned from the backend otherwise. `retrieveResponsePayload` feels like it's only retrieving from cache... Do you agree?

This made me think if we should move the logic of the function (checking for the response code) to HTTPClient. Although I would really like to isolate all this logic in `ETagManager` since it's logic related to ETags.

Reply:
yeah, that's tough. I agree that `retrieveResponsePayload` also doesn't solve the issue, since it does seem like you're retrieving it from cache. 

I can't think of a name that doesn't end up like one of those insane Cocoa names, i.e.: `getResponseFromCacheIfAbleOtherwiseParseResult`. 
But perhaps that's indicative of the method doing too much. 

@beylmk thoughts on this?

Reply:
yeah, i also had a little trouble understanding this the first time i read it, but i see how tricky the naming was. 

you could just call getStoredResult and storeResult from the HTTPClient directly, and maybe make a functions isResponseModified/shouldStoreResult in the eTagManager to still somewhat blackbox the logic from the HTTP Client?

so (in HTTPCLient performREquest) something like
```suggestion

                if (eTagManager.isResultChanged(request) && shouldStoreResult(result) {
            eTagManager.storeResult()
        } else {
            result = eTagManager.getResult()
        }
```

Reply:
I am going to follow your recommendations and refactor this bit. The more I look into this, the less I like the fact that ETagManager kinda mutates stuff.